functions/source/ecr-image-action-event-filtering/ecr_image_action_event_filtering_lambda_function.py

```diff
@@ -40,6 +40,12 @@ def lambda_handler(event, context):
     image_digest = event['detail']['image-digest']
     image_tag = event['detail'].get('image-tag',"")
 
+    # For SOCi V2, we create a new image index and tag it with a "-soci" suffix tag.
+    # This repo tag generates a new PUSH event and will retrigger the soci index builder.
+    # To prevent this recursive processing, skip image tag that contains "soci"
+    if "soci" in image_tag:
```

Comment:
Should we be more strict about this matching and specifically looks for a `-soci` suffix rather than soci anywhere in the string? I think as this could cause weird false positives that trip up customers.